qa/workunits/rbd/rbd_mirror.sh

```diff
@@ -1,26 +1,52 @@
 #!/bin/sh -ex
 #
-# rbd_mirror_journal.sh - test rbd-mirror daemon in journal-based mirroring mode
+# rbd_mirror.sh - test rbd-mirror daemon in snapshot or journal mirroring mode
+#
+# Usage:
+#   RBD_MIRROR_MODE=journal rbd_mirror.sh
+#
+# Use environment variable RBD_MIRROR_MODE to set the mode
+# Available modes: snapshot | journal
 #
 # The scripts starts two ("local" and "remote") clusters using mstart.sh script,
 # creates a temporary directory, used for cluster configs, daemon logs, admin
 # socket, temporary files, and launches rbd-mirror daemon.
 #
 
-. $(dirname $0)/rbd_mirror_helpers.sh
+if [ "${#}" -gt 0 ]; then
+  echo "unnecessary arguments: ${@}"
+  exit 100
+fi
 
+if [ "${RBD_MIRROR_MODE}" != "snapshot" ] && [ "${RBD_MIRROR_MODE}" != "journal" ]; then
+  echo "unknown mode: ${RBD_MIRROR_MODE}"
+  echo "set RBD_MIRROR_MODE env variable, available modes: snapshot | journal"
+  exit 100
+fi
+
+DESC_PATTERN=""
+MIRROR_POOL_MODE=pool
+MIRROR_IMAGE_MODE=journal
+if [ "${RBD_MIRROR_MODE}" = "journal" ]; then
+  DESC_PATTERN='primary_position'
+else
+  MIRROR_POOL_MODE=image
+  MIRROR_IMAGE_MODE=snapshot
+fi
```

Comment:
The `MIRROR_POOL_MODE` and ` MIRROR_IMAGE_MODE` are used in the helper file only. So why don't do these definitions there?

Also (nit) we could avoid using `DESC_PATTERN` if defined and used a helper function `wait_for_replay_status_in_pool_dir` that would call `wait_for_status_in_pool_dir` with necessary pattern, depending on `RBD_MIRROR_MODE`.

Reply:
OK